app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = current_user.comments.new(
+      author_id: current_user.id,
+      post_id: @post.id,
+      text: comment_params
+    )
+
+    @comment.save if @comment.valid?
+    redirect_to user_post_path(current_user, @post.id)
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)[:text]
+  end
+end
```

Comment:
- Please make sure there should be a pair of routes in the `config/routes.rb` for creating comments
- **Hint**: When you run `rails routes`, you have to get this: `POST /posts/:post_id/comments(.:format) comments#creat` and `GET /posts/:post_id/comments/new(.:format) comments#new`  like what I paste in post controller
- 
**NB: This comment was supposed to be lined with `Route` file, unfortunetly you did not make the file visible**